langchain/src/vectorstores/usearch.ts

```diff
@@ -0,0 +1,202 @@
+    if (dv !== d) {
+      throw new Error(
+        `Vectors must have the same length as the number of dimensions (${d})`
+      );
+    }
+
+
+    const docstoreSize = this.index.size();
+    const documentIds = [];
+    for (let i = 0; i < vectors.length; i += 1) {
+      const documentId = uuid.v4();
+      documentIds.push(documentId);
+      const id = docstoreSize + i;
+      this.index.add(BigInt(id), new Float32Array(vectors[i]));
+      this._mapping[id] = documentId;
+      this.docstore.add({ [documentId]: documents[i] });
+    }
+    return documentIds;
+  }
+
+
+  async similaritySearchVectorWithScore(query: number[], k: number) {
+    const d = this.index.dimensions();
+    if (query.length !== d) {
+      throw new Error(
+        `Query vector must have the same length as the number of dimensions (${d})`
+      );
+    }
+    if (k > this.index.size()) {
+      const total = this.index.size();
+      console.warn(
+        `k (${k}) is greater than the number of elements in the index (${total}), setting k to ${total}`
+      );
+      // eslint-disable-next-line no-param-reassign
+      k = total;
+    }
+    const result = this.index.search(new Float32Array(query), k);
+
+
+    let return_list: [Document, number][] = [];
+    for (let i = 0; i < result.count; i++) {
+      const uuid = this._mapping[Number(result.keys[i])];
+      return_list.push([this.docstore.search(uuid), result.distances[i]]);
+    }
+
+
+    return return_list;
+  }
+
+
+  async save(directory: string) {
+    const fs = await import("node:fs/promises");
+    const path = await import("node:path");
+    await fs.mkdir(directory, { recursive: true });
+    await Promise.all([
+      this.index.save(path.join(directory, "usearch.index")),
+      await fs.writeFile(
+        path.join(directory, "docstore.json"),
+        JSON.stringify([
+          Array.from(this.docstore._docs.entries()),
+          this._mapping,
+        ])
+      ),
+    ]);
+  }
+
+
+  static async fromTexts(
+    texts: string[],
+    metadatas: object[] | object,
+    embeddings: Embeddings,
+    dbConfig?: {
+      docstore?: SynchronousInMemoryDocstore;
+    }
+  ): Promise<USearch> {
+    const docs: Document[] = [];
+    for (let i = 0; i < texts.length; i += 1) {
+      const metadata = Array.isArray(metadatas) ? metadatas[i] : metadatas;
+      const newDoc = new Document({
+        pageContent: texts[i],
+        metadata,
+      });
+      docs.push(newDoc);
+    }
+    return this.fromDocuments(docs, embeddings, dbConfig);
+  }
+
+
+  static async fromDocuments(
+    docs: Document[],
+    embeddings: Embeddings,
+    dbConfig?: {
+      docstore?: SynchronousInMemoryDocstore;
+    }
+  ): Promise<USearch> {
+    const args: USearchArgs = {
+      docstore: dbConfig?.docstore,
+    };
+    const instance = new this(embeddings, args);
+    await instance.addDocuments(docs);
+    return instance;
+  }
+
+
+  static async importUSearch(): Promise<{ Index: typeof Index }> {
```

Comment:
You can actually just import this normally:

https://github.com/hwchase17/langchainjs/blob/main/.github/contributing/INTEGRATIONS.md#creating-a-separate-entrypoint